src/Microsoft.Framework.Runtime.Abstractions/DiagnosticResult.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Framework.Runtime
+{
+    public class DiagnosticResult
+    {
+        public static readonly DiagnosticResult Successful = new DiagnosticResult(success: true, diagnostics: Enumerable.Empty<DiagnosticMessage>());
+
+        public bool Success { get; }
+
+        public IEnumerable<DiagnosticMessage> Diagnostics { get; }
+
+        public DiagnosticResult(bool success, IEnumerable<DiagnosticMessage> diagnostics)
```

Comment:
I've mostly seen constructors after properties, maybe I'm looking at the wrong code... General C# style that I've seen in practice has constructors after properties but that's just anecdotal. Is there a specific style guide we use?